packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -604,12 +604,18 @@ class SkiaGoldClient {
 
   /// Returns a boolean value to prevent the client from re-authorizing itself
   /// for multiple tests.
-  bool _clientIsAuthorized() {
+  Future<bool> clientIsAuthorized() async {
     final File authFile = workDirectory?.childFile(fs.path.join(
       'temp',
       'auth_opt.json',
     ));
-    return authFile.existsSync();
+
+    if(await authFile.exists()) {
```

Comment:
Can we just avoid writing this file to a location that gets cached between CI runs?

Reply:
I would like to do that, I filed #50148 to keep up with it and hopefully just incorporate it into the refactoring I'm doing for #49837 and #49749. I think I am going to have to change where all of the files are being managed, json/images/etc, not just this file, which will take a bit more doing. I wanted to unblock others in the meantime.

Reply:
Can we add a TODO with a link to those bugs/PRs then?

Reply:
Unless, do you think just using the CIRRUS_WORKING_DIR as the skia client's work directory would do it? I don't know yet where Cirrus draws the line on caching. I thought it wasn't caching these originally. 😅

Reply:
On macs we don't do any caching.  You can find out which directories are caching by looking for task names postfixed with _cache.

...although now that I look at it, it seems like the directory this is in doesn't get cached.  Which directory is it again?

Reply:
That's what I thought.. here's one:
```
Authentication set up in directory
/tmp/flutter sdk/bin/cache/pkg/skia_goldens1880/packages/flutter/test/widgets/temp
```

Reply:
ah..
from .cirrus.yml for linux shards:
```
flutter_pkg_cache:
    folder: bin/cache/pkg
    fingerprint_script: echo $OS; cat bin/internal/*.version
```

Reply:
I might be able to get away with just using Directory.systemTemp for when we are on Cirrus. That should clean up after itself. I don't want to hide golden files when folks are using them elsewhere.

Reply:
Yeah, we're caching the pkg directory for sure.

Reply:
The goldens repo used to invalidate the cache since it had a version file.